src/components/AudioLevelIndicator/AudioLevelIndicator.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useRef } from 'react';
+import { AudioTrack, LocalAudioTrack, RemoteAudioTrack } from 'twilio-video';
+import MicOff from '@material-ui/icons/MicOff';
+import useIsTrackEnabled from '../../hooks/useIsTrackEnabled/useIsTrackEnabled';
+
+let clipId = 0;
+const getUniqueClipId = () => clipId++;
+
+// @ts-ignore
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+let audioContext: AudioContext;
+
+function AudioLevelIndicator({
+  size,
+  audioTrack,
+  background,
+}: {
+  size?: number;
+  audioTrack?: AudioTrack;
+  background?: string;
+}) {
+  const SIZE = size || 24;
+  const ref = useRef<SVGRectElement>(null);
+  const isTrackEnabled = useIsTrackEnabled(audioTrack as LocalAudioTrack | RemoteAudioTrack);
+
+  useEffect(() => {
+    const SVGClipElement = ref.current;
+    let animationFrameID: number;
+
+    if (audioTrack && isTrackEnabled && SVGClipElement) {
+      audioContext = audioContext || new AudioContext();
+      const stream = new MediaStream();
+      stream.addTrack(audioTrack.mediaStreamTrack.clone());
+      const audioSource = audioContext.createMediaStreamSource(stream);
+
+      const analyser = audioContext.createAnalyser();
+      analyser.smoothingTimeConstant = 0.5;
+      analyser.fftSize = 1024;
+
+      audioSource.connect(analyser);
+
+      function render() {
+        animationFrameID = window.requestAnimationFrame(() => {
+          const array = new Uint8Array(analyser.frequencyBinCount);
+          analyser.getByteFrequencyData(array);
+          let values = 0;
+
+          const length = array.length;
+          for (let i = 0; i < length; i++) {
+            values += array[i];
+          }
+
+          const volume = Math.min(21, Math.max(0, Math.log10(values / length / 3) * 14));
```

Comment:
It feels like getting volume levels should be part of the SDK. Is this API not supported yet?

Reply:
@charliesantos Currently it is not. We are thinking of adding it though.